src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -456,7 +459,10 @@ BlobMetadata buildAndUpload(String filePath, List<List<ChannelData<T>>> blobData
                 .setChunkLength(compressedChunkLength)
                 .setChannelList(result.channelsMetadataList)
                 .setChunkMD5(md5)
-                .setEncryptionKeyId(firstChannel.getEncryptionKeyId())
+                .setEncryptionKeyId(
+                    bdecVersion == Constants.BdecVersion.THREE
+                        ? 0
```

Comment:
We might want to pull this out as a constant of `NO_ENCRYPTION_KEY_ID` so that it is obvious why `0` is being used here.